go/oasis-node/cmd/common/metrics/metrics.go

```diff
@@ -237,20 +228,20 @@ func (s *pushService) initPusher(isReinit bool) {
 func newPushService() (service.BackgroundService, error) {
 	svc := &pushService{
 		BaseBackgroundService: *service.NewBaseBackgroundService("metrics"),
-		addr:                  viper.GetString(CfgMetricsAddr),
-		jobName:               viper.GetString(CfgMetricsJobName),
-		labels:                viper.GetStringMapString(CfgMetricsLabels),
-		interval:              viper.GetDuration(CfgMetricsInterval),
-		rsvc:                  newResourceService(viper.GetDuration(CfgMetricsInterval)),
+		addr:                  config.GlobalConfig.Metrics.Address,
+		jobName:               config.GlobalConfig.Metrics.JobName,
+		labels:                config.GlobalConfig.Metrics.Labels,
+		interval:              config.GlobalConfig.Metrics.Interval,
+		rsvc:                  newResourceService(config.GlobalConfig.Metrics.Interval),
 		stopCh:                make(chan struct{}),
 		quitCh:                make(chan struct{}),
 	}
 
 	if svc.jobName == "" {
-		return nil, fmt.Errorf("metrics: %s required for push mode", CfgMetricsJobName)
+		return nil, fmt.Errorf("metrics: job_name required for push mode")
```

Comment:
Added to config validation.
But I think the existing text here is better, because it actually tells you which field is missing.

Reply:
> existing text here is better

Maybe you're right. I just thought that generalization is a bit better because then one can rename `job_name` to `jobname` without any extra work.